src/bootstrap/config.rs

```diff
@@ -2004,7 +2004,8 @@ impl Config {
                 .unwrap();
         if !(source_version == rustc_version
             || (source_version.major == rustc_version.major
-                && source_version.minor == rustc_version.minor + 1))
+                && (source_version.minor == rustc_version.minor
+                    || source_version.minor == rustc_version.minor + 1)))
```

Comment:
> e.g. Rust 1.71.1 could not be built by Rust 1.71.0 because it is neither an exact version match, or the previous version.

Shouldn't we handle `.patch` here?

Reply:
Do you expect a scenario where a future patchlevel cannot build an older patchlevel?  My understanding is that the compatibility requirement is `X.Y.* || X.Y + 1.*`, where `Y + 1` is current.

Reply:
I missed this line:

https://github.com/rust-lang/rust/blob/31a81a08786826cc6e832bd0b49fb8b934e29648/src/bootstrap/config.rs#L2010